sdks/java/harness/src/main/java/org/apache/beam/fn/harness/logging/BeamFnLoggingClient.java

```diff
@@ -122,7 +124,9 @@ public BeamFnLoggingClient(
     logRecordHandler.setLevel(Level.ALL);
     logRecordHandler.setFormatter(DEFAULT_FORMATTER);
     logRecordHandler.executeOn(options.as(ExecutorOptions.class).getScheduledExecutorService());
-    outboundObserver = (CallStreamObserver<BeamFnApi.LogEntry.List>) stub.logging(inboundObserver);
+    outboundObserver =
+        new DirectStreamObserver<BeamFnApi.LogEntry.List>(
+            phaser, (CallStreamObserver<BeamFnApi.LogEntry.List>) stub.logging(inboundObserver));
```

Comment:
I was thinking the logging for slow write would be nice to show that it is falling behind due to runner not consuming.

Reply:
Checked performance impact, seems like we get a net win since we do the readiness checking less often

before:
```
Benchmark                                                           Mode  Cnt       Score       Error  Units
BeamFnLoggingClientBenchmark.testLogging                           thrpt   15  415642.810 ± 17448.195  ops/s
BeamFnLoggingClientBenchmark.testLoggingWithAllOptionalParameters  thrpt   15  380592.277 ±  8459.947  ops/s
```

after:
```
Benchmark                                                           Mode  Cnt       Score       Error  Units
BeamFnLoggingClientBenchmark.testLogging                           thrpt   15  418828.259 ± 15726.823  ops/s
BeamFnLoggingClientBenchmark.testLoggingWithAllOptionalParameters  thrpt   15  402892.740 ± 10908.775  ops/s
```